server/swap/swap.go

```diff
@@ -983,6 +1009,9 @@ func (s *Swapper) readMatches(matchSets []*order.MatchSet) []*matchTracker {
 	// reallocs.
 	matches := make([]*matchTracker, 0, len(matchSets))
 	for _, matchSet := range matchSets {
+		if matchSet.Taker.Type() == order.CancelOrderType {
+			continue
+		}
 		for _, match := range matchSet.Matches() {
```

Comment:
Note that this was the problem for cancel orders (`matchset.Matches()` panics since Amounts and Rates are not set for cancel order Matches).  However, I see that the output slice from `readMatches` must include the cancel orders too, so I will special case this instead of simply `continue`.

Reply:
Now setting Amounts and Rates for cancel order MatchSets too: 3958263